src/packagedcode/rubygems.py

```diff
@@ -648,92 +628,163 @@ def normalize(gem_data, known_fields=known_fields):
 
 
 def parse_spec(location):
-    pass
-
+    """
+    Returns a dictionary containing .gemspec file data.
+    """
+    file_data = []
+    with io.open(location, "r", encoding="utf-8") as f:
+        file_data = f.read().splitlines()
+
+    # defaults fields of .gemspec file
+    gemspec_data = {
+        "name": None,
+        "version": None,
+        "authors": None,
+        "email": None,
+        "summary": None,
+        "description": None,
+        "homepage": None,
+        "license": None,
+        "dependencies": None
+        }
 
-class GemSpec(object):
+    dependencies = OrderedDict()
+
+    for individual in file_data:
+        # update the value of name
+        if '.name ' in individual and 'basename' not in individual:
+            # >>> s = 'spec.name = "abc"'
+            # >>> name = s.split('= ')
+            # >>> name
+            # ['spec.name ', '"abc"']
+            # >>> name = name[1].strip('\'"%q{}')
+            # >>> name
+            # abc
+            name = individual.split('= ')
+            name = name[1].strip('\'"%q{}')
```

Comment:
What if there is no `=`?
```
>>> 'foo'.split('= ')[1]
Traceback (most recent call last):
  File "<stdin>", line 1, in <module>
IndexError: list index out of range
```
What if the name starts with `q`?
```
>>> 'quiet'.strip('\'"%q{}')
'uiet'
```
Your should use a partition() or a .replace() instead for safety.